src/Compilers/CSharp/Portable/Binder/Binder_TupleOperators.cs

```diff
@@ -0,0 +1,386 @@
+
+                return BindTupleBinaryOperatorNestedInfo(node, kind,
+                    GetTupleArgumentsOrPlaceholders(left), GetTupleArgumentsOrPlaceholders(right),
+                    leftNullable || rightNullable, diagnostics);
+            }
+
+            return BindTupleBinaryOperatorSingleInfo(node, kind, left, right, diagnostics);
+        }
+
+        private TupleBinaryOperatorInfo BindTupleBinaryOperatorSingleInfo(BinaryExpressionSyntax node, BinaryOperatorKind kind,
+            BoundExpression left, BoundExpression right, DiagnosticBag diagnostics)
+        {
+            TypeSymbol leftType = left.Type;
+            TypeSymbol rightType = right.Type;
+
+            if ((object)leftType != null && leftType.IsDynamic() || (object)rightType != null && rightType.IsDynamic())
+            {
+                return BindTupleDynamicBinaryOperatorSingleInfo(node, kind, left, right, diagnostics);
+            }
+
+            LookupResultKind resultKind;
+            BinaryOperatorAnalysisResult best = this.BinaryOperatorOverloadResolution(kind, left, right, node, diagnostics, out resultKind, out var _);
+
+            MethodSymbol resultMethod = null;
+            BinaryOperatorKind resultOperatorKind;
+            TypeSymbol convertedLeftType = null;
+            TypeSymbol convertedRightType = null;
+            TypeSymbol returnType;
+            bool hasErrors;
+
+            if (!best.HasValue)
+            {
+                resultOperatorKind = kind;
+                returnType = CreateErrorType();
+                hasErrors = true;
+            }
+            else
+            {
+                bool leftNull = left.IsLiteralNull();
+                bool rightNull = right.IsLiteralNull();
+
+                var signature = best.Signature;
+
+                bool isObjectEquality = signature.Kind == BinaryOperatorKind.ObjectEqual || signature.Kind == BinaryOperatorKind.ObjectNotEqual;
+
+                resultOperatorKind = signature.Kind;
+                resultMethod = signature.Method;
+                convertedLeftType = signature.LeftType;
+                convertedRightType = signature.RightType;
+                returnType = signature.ReturnType; // PROTOTYPE(tuple-equality) Do we need a special case for nullable equality? (see BindSimpleBinaryOperator)
+
+                HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+                hasErrors = isObjectEquality && !BuiltInOperators.IsValidObjectEquality(Conversions, leftType, leftNull, rightType, rightNull, ref useSiteDiagnostics);
+                diagnostics.Add(node, useSiteDiagnostics);
+            }
+
+            if (convertedLeftType is null || convertedRightType is null)
+            {
+                convertedLeftType = convertedLeftType ?? leftType ?? CreateErrorType();
+                convertedRightType = convertedRightType ?? rightType ?? CreateErrorType();
+                hasErrors = true;
```

Comment:
Can we hit this case other than when `!best.HasValue`? If not, perhaps set `converted*Type = CreateErrorType();` in that `if` and removing this `if`.